src/mlpack/core/data/load_impl.hpp

```diff
@@ -410,6 +410,110 @@ bool LoadCategorical(const std::string& filename,
   return true;
 }
 
+template<typename MatType>
+bool Load(const std::vector<std::string>& filenames,
+          MatType& matrix,
+          const TextOptions& opts)
+{
+  TextOptions copyOpts(opts);
+  return Load(filenames, matrix, copyOpts);
+}
+
+template<typename MatType>
+bool Load(const std::vector<std::string>& filenames,
+          MatType& matrix,
+          TextOptions& opts)
+{
+  bool success = false;
+  MatType tmp;
+  arma::field<std::string> firstHeaders;
+  if (filenames.empty())
+  {
+    if (opts.Fatal())
+    {
+      Log::Fatal << "Load(): Please specify the filenames to be loaded!"
+          << std::endl;
+    }
+    else
+    {
+      Log::Warn << "Load(): Please specify the filenames to be loaded!"
+          << std::endl;
+      return false;
+    }
+  }
+  for (size_t i = 0; i < filenames.size(); ++i)
+  {
+    success = Load(filenames.at(i), matrix, opts);
+    if (opts.HasHeaders())
+    {
+      if (i == 0)
+        firstHeaders = opts.Headers();
+      else
+      {
+        arma::field<std::string>& headers = opts.Headers();
+        for (size_t j = 0; j < headers.size(); ++j)
+        {
+          if (firstHeaders.at(0) != headers.at(0))
+            Log::Warn << "Load(): Headers from different CSV does not match!"
+                << std::endl;
```

Comment:
So what is the action we want to take here?  I believe you want loading to fail entirely if the headers don't match, instead of only loading some files.  But if you want to do that, we would need to either issue this message to `Fatal` or `Warn` depending on the setting of `Fatal()`, and we would need to `return false` and clear `matrix`.  (If loading fails, then we should make sure the matrix is empty.)

I'd also suggest printing a more verbose error message to make it clear what has gone wrong, something like this:

```
Load(): header column 5 in 'filename_2.csv' ("column_name") does not match 'filename_1.csv' ("orig_column_name")!  Loading failed.
```

This will give the user enough information to know exactly what's wrong.